src/Catalog/CatalogItem.cs

```diff
@@ -10,11 +10,25 @@ namespace NuGet.Services.Metadata.Catalog
 {
     public abstract class CatalogItem
     {
+        Uri _baseAddress;
+
         public DateTime TimeStamp { get; set; }
 
         public Guid CommitId { get; set; }
 
-        public Uri BaseAddress { get; set; }
+        public Uri BaseAddress
+        {
+            get
+            {
+                return AliasesAddress(_baseAddress, CustomDomain);
```

Comment:
It will be a small perf impact. If changed in the suggested way then the set of the base address needs to happen after the set of custom domain. In the current way the get will use the data at the current time.